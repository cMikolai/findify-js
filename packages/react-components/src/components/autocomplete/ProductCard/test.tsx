import React from 'react'
import ProductCardView from './view'
import { shallow } from 'enzyme'
import { get as _get } from 'lodash'

const itemMock = {
  product_url: "https://foo.bar",
  image_url: "https://lorempicsum.net/100/100",
  description: "A description for item",
  thumbnail_url: "https://lorempicsum.net/50/50",
  title: "Product Title",
  price: "9.99",

  get(key) {
    return this[key]
  }
}

const configMock = {
  title: {
    display: true,
    lines: 1
  },
  description: {
    display: true,
    lines: 2
  },
  price: {
    display: true
  },
  get(key) {
    return _get(this, key)
  }
}

describe('ProductCard view', () => {
  it('renders correctly', () => {
    expect(shallow(
      <ProductCardView
        item={itemMock}
        config={configMock}
      />
    )).toMatchSnapshot()
  })
})
