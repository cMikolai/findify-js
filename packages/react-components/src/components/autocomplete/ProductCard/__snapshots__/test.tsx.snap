// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard view renders correctly 1`] = `
<a
  className="root"
  href="https://foo.bar"
>
  <div
    className="imageWrap"
  >
    <withPropsOnChange(withStateHandlers(withPropsOnChange(mapProps(onlyUpdateForKeys(Component)))))
      alt="Product Title"
      className="image"
      src="https://lorempicsum.net/100/100"
    />
  </div>
  <div
    className="content"
  >
    <Title
      text="Product Title"
    />
    <Description
      text="A description for item"
    />
  </div>
</a>
`;
